pages/employee/index.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+
+    <UForm :validate="validate" :state="employee" class="space-y-4" @submit="submitEmployee" @error="onError">
+        <div class="w-96">
+            <div class="mb-2">
+                <UFormGroup label="First Name" name="first_name">
+                    <UInput placeholder="Enter First Name" v-model="employee.first_name" />
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UFormGroup label="Last Name" name="last_name">
+                    <UInput placeholder="Enter Last Name" v-model="employee.last_name" />
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UFormGroup label="Mobile" name="mobile">
+                    <UInput placeholder="Enter Mobile" v-model="employee.mobile" :maxlength="10"
+                        @input="allowNumericValue" />
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UFormGroup label="Pincode" name="pincode">
+                    <UInput placeholder="Enter Pincode" :maxlength="6" v-model="employee.pincode" @change="checkPincode"
+                        @input="allowNumericValue" />
+                    <span v-if="pincodeError" class="text-red-500">{{ pincodeError }}</span>
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UFormGroup name="Employment Type" label="Employment Type">
+                    <UInputMenu v-model="employee.employe_type" :options="employeeTypes" />
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UFormGroup :label="incomeTitle" name="income">
+                    <UInput :placeholder="`Enter ${incomeTitle}`" v-model="employee.income" />
+                </UFormGroup>
+            </div>
+            <div class="mb-2">
+                <UButton type="submit">
+                    Submit
+                </UButton>
+            </div>
+        </div>
+    </UForm>
+</template>
+
+<script setup lang="ts">
+import type { FormError, FormErrorEvent, FormSubmitEvent } from '#ui/types'
+const employeeTypes = ['Business', 'Salaried']
+const employee = ref({
+    first_name: undefined,
+    last_name: undefined,
+    mobile: undefined,
+    pincode: undefined,
+    employe_type: employeeTypes[0],
+    income: undefined,
```

Comment:
Adding typescript support for this would help in dev. Creating an interface for the employee form fields.

Also, would recommend using either empty string values or null instead of undefined as our initial values.

Reply:
Resolved